### DISCUSSION TITLE
Query DynamoDB

### DISCUSSION THREAD

**Discussion by PRRPCHT** [Budibase/budibase#4477] [Help] [answered]
Hello,

Does anyone have an example on how to set a DynamoDB query? I can't figure out what goes where...

Thanks!

**Comment by mike12345567** [comment, marked as answer]
Hi @Reiep - currently, the DynamoDB query makes use of the AWS Dynamo Document Client API, the documentation for this can be found here: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html

For example, I've setup a local test Dynamo database and written to a table using the following query: 
![image](https://user-images.githubusercontent.com/4407001/153953679-663c9bcf-4abf-473b-858a-7d92cb31e46a.png)

Essentially the query needs to contain the JSON config requirement for the various operations that the document client can perform.

The operations currently supported are as follows:
1. Function: Create - DynamoDB `put`
2. Function: Read - DynamoDB `query`
3. Function: Describe - this is not a Document client operation, it doesn't requirement any other info other than the table to describe.
4. The functions Scan, Get, Update and Delete all match up directly with their Dynamo counterparts.
5. The transaction and batch operations are currently not supported.

Other than this, all you need to do is use the bindings to get parameters into your queries, you can see examples of how to do this for a query based on the [REST query documentation](https://docs.budibase.com/building-apps/data/tables/rest/rest-bindings-example).

In the future if there is enough interest there maybe an improvement to the DynamoDB integration to work in a more integrated way, like how our [SQL datasources](https://docs.budibase.com/building-apps/data/tables/postgresql-mysql-and-sql-server) work - but this isn't currently on the roadmap!

**Comment by PRRPCHT** [comment]
Thanks for the detailed answer, I'll give it a more thorough look.
